src/SFA.DAS.ApprenticeCommitments.Web/Exceptions/DomainValidationException.cs

```diff
@@ -1,21 +1,31 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Runtime.Serialization;
 
 namespace SFA.DAS.ApprenticeCommitments.Web.Exceptions
 {
-    internal class DomainValidationException : Exception
+    [Serializable]
+    public class DomainValidationException : Exception
     {
         public List<ErrorItem> Errors { get; }
 
         public DomainValidationException(List<ErrorItem> errors) : base("DomainValidation Exception")
         {
             Errors = errors;
         }
+
+        protected DomainValidationException(SerializationInfo info, StreamingContext context)
+        : base(info, context)
+        {
+            _ = info ?? throw new ArgumentNullException(nameof(info));
```

Comment:
> 
> 
> this assignment is unusual, why not say if(info == null) throw ......

It's a (relatively?) common way of doing the null check - MS calls it "[standalone discard](https://docs.microsoft.com/en-us/dotnet/csharp/discards#a-standalone-discard)"

> Also it's quite unusual to throw an exception within an exception ctor. Argument for failing at this point would be so that we do not hide a deserialiser error

Wouldn't we just throw an implicit `NullReferenceException` on the next line anyway when `info` is derefenced?  I think it's probably more useful to have the exception saying what is null before that, and I don't think if changes the exception-within-exception semantics does it?